### PR TITLE
properly include babel for Mocha's sake

### PR DESCRIPTION
**Pull request by amitport**

### CODE REVIEW

**Comment by SBoudrias** [comment]
Can you add a unit test

**Comment by amitport** [comment]
I'm really not familiar with yeoman testing and its helpers, and I'm not sure how to test this in any case...

**Comment by arthurvr** [comment]
Also please squash your commits.

> I'm really not familiar with yeoman testing and its helpers, and I'm not sure how to test this in any case...

I agree that this isn't the easiest thing to test. Maybe just assert the file content of `package.json` and `gulpfile.js` (and assert that it's not there when the babel option is disabled)

File: generators/gulp/index.js [hunk 1]
```diff
@@ -33,7 +33,8 @@ module.exports = generators.Base.extend({
         'gulp-jshint': '^1.5.3',
         'gulp-mocha': '^2.0.0',
         'gulp-plumber': '^1.0.0',
-        'jshint-stylish': '^1.0.0'
+        'jshint-stylish': '^1.0.0',
+        'babel': '^5.6.5'
```
Comment (arthurvr) [inline generators/gulp/index.js]: this is already included in line 45

File: generators/gulp/templates/gulpfile.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ gulp.task('test', ['pre-test'], function (cb) {
 
   gulp.src('test/**/*.js')
     .pipe(plumber())
-    .pipe(mocha({reporter: 'spec'}))
+    .pipe(mocha({reporter: 'spec', require: ['babel/register']}))
```
Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: this should only be here if the `babel` option is turned on, right?

Comment (amitport) [inline generators/gulp/templates/gulpfile.js]: right

File: generators/gulp/index.js (hunk 1 above)
Comment (amitport) [inline generators/gulp/index.js]: removed line

File: generators/gulp/templates/gulpfile.js (hunk 2 above)
Comment (amitport) [inline generators/gulp/templates/gulpfile.js]: done

File: generators/gulp/templates/gulpfile.js [hunk 3]
```diff
@@ -42,7 +42,8 @@ gulp.task('test', ['pre-test'], function (cb) {
 
   gulp.src('test/**/*.js')
     .pipe(plumber())
-    .pipe(mocha({reporter: 'spec'}))
+    .pipe(mocha({reporter: 'spec'<% if (babel) { %>,
+      require: ['babel/register']<% } %>}))
```
Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: This is going to look a little strange:

```js
    .pipe(mocha({reporter: 'spec',
       require: ['babel/register']}))
```

it should be eighter all on one line, or all options on different lines.

**Comment by amitport** [comment]
> Also please squash your commits.

hmmm... also not that familiar with git. Do you mean that I should run:

`git rebase -i 3574ef5` (where `3574ef5` is commit I forked from)
`git push origin master` (where `master` is my fork)

?

**Comment by SBoudrias** [comment]
@amitport Yes, add 's' in front of every commit in the rebase screen to merge them

**Comment by amitport** [comment]
rebase screen?

**Comment by arthurvr** [comment]
@amitport GitHub has excellent documentation: https://help.github.com/articles/about-git-rebase/ ;)

**Comment by amitport** [comment]
done (that was fun...)

File: generators/gulp/templates/gulpfile.js (hunk 3 above)
Comment (stevemao) [inline generators/gulp/templates/gulpfile.js]:
```js
.pipe(mocha({
       reporter: 'spec',
<% if (babel) { -%>
       require: ['babel/register']
<% } -%>
}))
```

Comment (arthurvr) [inline generators/gulp/templates/gulpfile.js]: @stevemao @amitport Using that snippet we end up with a trailing comma (and notice that it's the wrong indentation)

Comment (stevemao) [inline generators/gulp/templates/gulpfile.js]: Yeah, just muck with it. mine just an idea what you could do (take advantage with the new templating system and the styling of the object).

**Comment by arthurvr** [comment]
> done (that was fun...)

Still needs the test and the reformatting :)

**Comment by amitport** [comment]
The generated Mocha **does not work at all** without this patch. The appropriate test that will prevent such issues in the future is a Mocha sanity/liveliness test. That's way above the scope of this change and the project survived without it so far. Should open a separate issue. 
As for the formatting I'll add a fix later today.

**Comment by arthurvr** [comment]
> The appropriate test that will prevent such issues in the future is a Mocha sanity/liveliness test.

The only test we asked for is a simple assert your `if (babel) {}` in the template works fine. That's it.

**Comment by SBoudrias** [comment]
@arthurvr Would you be good enough to follow up with a PR adding unit test for this?

**Comment by SBoudrias** [comment]
@amitport Thanks for taking care of this.

**Comment by arthurvr** [comment]
I would :) Gonna do that hopefully this evening :wink:

**Pull request merged**